rasterio/rio/shapes.py

```diff
@@ -136,6 +139,8 @@ def __call__(self):
                         src.width % sampling, src.height % sampling)
                     # And follow by scaling.
                     transform *= Affine.scale(float(sampling))
+                    shape = (int(math.ceil(src.height / sampling)),
+                             int(math.ceil(src.width / sampling)))
```

Comment:
Done, plus calculating independent sampling factors for x and y dimensions to handle the translation properly (I think).